Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/SANS/SANSSave.py

```diff
@@ -204,12 +233,56 @@ def PyExec(self):
                 raise e
         progress.report("Finished saving workspace to files.")
 
+    def _extract_metadata(self, workspace) -> dict[str:Any]:
+        additional_properties = {}
+        sample = workspace.sample()
+        maybe_geometry = convert_to_shape(sample.getGeometryFlag())
+        additional_properties["SampleHeight"] = sample.getHeight()
+        additional_properties["SampleWidth"] = sample.getWidth()
+        additional_properties["SampleThickness"] = sample.getThickness()
+
+        instrument = SANSInstrument_string_as_key_NoInstrument[workspace.getInstrument().getName()]
+        additional_properties["DetectorNames"] = ",".join(get_detector_names_from_instrument(instrument))
+
+        if maybe_geometry is not None:
+            additional_properties["Geometry"] = maybe_geometry.value
+
+        additional_properties["BackgroundSubtractionWorkspace"] = self.getProperty("BackgroundSubtractionWorkspace").value
+        additional_properties["BackgroundSubtractionScaleFactor"] = self.getProperty("BackgroundSubtractionScaleFactor").value
```

Comment:
Can you either rename this function and bring:
```
        additional_properties["Transmission"] = transmission
        additional_properties["TransmissionCan"] = transmission_can
            if transmission:
                transmission = get_zero_error_free_workspace(transmission)
            if transmission_can:
                transmission_can = get_zero_error_free_workspace(transmission_can)
```
inside, or move these out of the function.